daq/host.py

```diff
@@ -344,20 +359,20 @@ def _base_start(self):
             self._monitor_cleanup()
             self._monitor_error(e)
 
-    def _monitor_cleanup(self, forget=True):
+    def _monitor_cleanup(self):
         if self._tcp_monitor:
             LOGGER.info('Target port %d monitor scan complete', self.target_port)
-            if forget:
+            if self._tcp_monitor.stream() and not self._tcp_monitor.stream().closed:
```

Comment:
I'm still not a fan of these explicit "close if not closed" checks... I would want to know the case when we don't know in advance that the stream is closed already (it can happen, but should be the exception corner-case not the default behavior). I think there's this one and then one in gateway.py.  This was the reason behind the forget argument.  Anyway -- I took them out and I'm running the tests again, so we can see what happens...

Reply:
Ok, created https://github.com/henry54809/daq/pull/7 which makes the forget calls explicit, but keeps the check-closed-state logic with an assert.  Best of both worlds, maybe.